core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1521,8 +1521,15 @@ abstract class RDD[T: ClassTag](
     }
 
     checkpointData match {
-      case Some(reliable: ReliableRDDCheckpointData[_]) => logWarning(
-        "RDD was already marked for reliable checkpointing: overriding with local checkpoint.")
+      case Some(reliable: ReliableRDDCheckpointData[_]) =>
```

Comment:
I suspect this is a good restriction, but cc @tdas regarding the semantics.

I feel like this code construct is getting hard to read. I don't feel strongly, but is `if (checkpointData.isDefined && isCheckpointed)` not simpler? then there are just two branches, one return point.